src/System.Net.Http/tests/UnitTests/DigestAuthenticationTests.cs

```diff
@@ -40,5 +40,16 @@ public static IEnumerable<object[]> DigestResponse_Challenge_TestData()
             yield return new object[] { "=value1,key2=,", s_emptyStringList, s_emptyStringList };
             yield return new object[] { "key1\tm= value1", s_emptyStringList, s_emptyStringList };
         }
+
+        [Theory]
+        [InlineData("realm=\"NetCore\", nonce=\"qMRqWgAAAAAQMjIABgAAAFwEiEwAAAAA\", qop=\"auth\", stale=false")]
```

Comment:
Can you add other tests verifying the cases where the TrySetDigestAuthToken must return false.. From the RFC, realm and nonce are required, so the above code should return null if realm is not found in parameters..

```
 digest-challenge  = 1#( realm | [ domain ] | nonce |
                          [ opaque ] |[ stale ] | [ algorithm ] |
                          [ qop-options ] | [auth-param] )
```